### ISSUE TITLE
Two `insert_many()` in a row raise a DB lock error

### ISSUE THREAD

**Issue by riccardomurri** [pudo/dataset#163]
According to my understanding of the docs, these two snippets of code
should be functionally equivalent, only the first should be faster
than the second::

        # 1. use `insert_many()`
        db['perfdata'].insert_many(dict(host=host, step=step, **row) for row in data)

        # 2. use `insert()` in a loop
        perfdata_tab = db['perfdata']
        for row in data:
            perfdata_tab.insert(dict(host=host, step=step, **row))

However, when this code is itself executed in a loop (`data` changes
at each iteration), the second version works correctly, whereas the
`insert_many()` dies with the following traceback::

        Traceback (most recent call last):
          File "./collant.py", line 708, in <module>
            cli()
          [...]
          File "./collant.py", line 624, in _load_perfdata_file
            db['perfdata'].insert_many(dict(host=host, step=step, **row) for row in data)
          File "/home/rmurri/.virtualenvs/collant/local/lib/python2.7/site-packages/dataset/persistence/table.py", line 106, in insert_many
            _process_chunk(chunk)
          File "/home/rmurri/.virtualenvs/collant/local/lib/python2.7/site-packages/dataset/persistence/table.py", line 95, in _process_chunk
            self.table.insert().execute(chunk)
          File "/home/rmurri/.virtualenvs/collant/local/lib/python2.7/site-packages/sqlalchemy/sql/base.py", line 386, in execute
            return e._execute_clauseelement(self, multiparams, params)
          File "/home/rmurri/.virtualenvs/collant/local/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1998, in _execute_clauseelement
            return connection._execute_clauseelement(elem, multiparams, params)
          File "/home/rmurri/.virtualenvs/collant/local/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1010, in _execute_clauseelement
            compiled_sql, distilled_params
          File "/home/rmurri/.virtualenvs/collant/local/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1146, in _execute_context
            context)
          File "/home/rmurri/.virtualenvs/collant/local/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1341, in _handle_dbapi_exception
            exc_info
          File "/home/rmurri/.virtualenvs/collant/local/lib/python2.7/site-packages/sqlalchemy/util/compat.py", line 200, in raise_from_cause
            reraise(type(exception), exception, tb=exc_tb, cause=cause)
          File "/home/rmurri/.virtualenvs/collant/local/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1116, in _execute_context
            context)
          File "/home/rmurri/.virtualenvs/collant/local/lib/python2.7/site-packages/sqlalchemy/engine/default.py", line 447, in do_executemany
            cursor.executemany(statement, parameters)
        sqlalchemy.exc.OperationalError: (sqlite3.OperationalError) database is locked [SQL: u'INSERT INTO perfdata (nfs_4cd_symlink, nfs_2sd_remove, nfs_3sd_setattr, nfs_2cd_create, net_rxkbtot, nfs_3cd_pathconf, nfs_2sd_create, nfs_2cd_link, nfs_2cd_mkdir, nfs_3cd_fsstat, nfs_3cd_readlink, nfs_3cd_readdir, nfs_2sd_readlink, nfs_3cd_write, nfs_2cd_write, cpu_runtot, cpu_totl_percent, cpu_guestn_percent, timestamp, nfs_2sd_setattr, nfs_3sd_fsinfo, nfs_2sd_readdir, nfs_3cd_link, nfs_4cd_rename, nfs_2sd_lookup, net_txpkttot, nfs_3sd_fsstat, nfs_2sd_read, nfs_3sd_mkdir, nfs_3cd_commit, nfs_3sd_null, nfs_4sd_read, nfs_2cd_read, cpu_blktot, nfs_4cd_remove, nfs_3sd_link, nfs_3sd_create, nfs_4sd_setattr, nfs_3cd_access, nfs_4cd_getattr, nfs_3sd_readlink, nfs_4cd_setattr, nfs_2cd_setattr, cpu_sys_percent, nfs_4sd_getattr, nfs_4sd_access, nfs_3cd_rmdir, nfs_4sd_create, nfs_3cd_rename, cpu_user_percent, nfs_3sd_symlink, nfs_2cd_null, nfs_3sd_getattr, nfs_2cd_symlink, nfs_2sd_getattr, cpu_ctx_per_second, nfs_4cd_readdir, nfs_4cd_fsinfo, nfs_2sd_rmdir, step, nfs_3sd_remove, nfs_4sd_lookup, nfs_4cd_readlink, nfs_3sd_rmdir, nfs_2cd_rmdir, nfs_2sd_symlink, nfs_2sd_mkdir, nfs_4sd_commit, nfs_3sd_pathconf, cpu_proc_per_second, nfs_3cd_null, cpu_guest_percent, nfs_4cd_access, nfs_4sd_readdir, nfs_4cd_pathconf, nfs_4cd_create, nfs_3cd_readdirplus, nfs_4sd_write, nfs_3sd_lookup, net_rxerrstot, nfs_4cd_read, nfs_3cd_mknod, nfs_3cd_lookup, cpu_procque, cpu_idle_percent, nfs_4sd_remove, nfs_3cd_create, cpu_procrun, nfs_3sd_access, cpu_wait_percent, nfs_2cd_lookup, cpu_l_avg15, net_rxpkttot, nfs_3cd_mkdir, nfs_3sd_mknod, cpu_nice_percent, nfs_2cd_readdir, nfs_3cd_read, nfs_2cd_getattr, nfs_2sd_write, nfs_4cd_commit, nfs_3sd_rename, nfs_3sd_write, nfs_2cd_rename, nfs_3sd_readdirplus, cpu_irq_percent, host, nfs_3sd_readdir, nfs_3cd_fsinfo, cpu_steal_percent, nfs_2cd_readlink, nfs_2sd_link, nfs_4cd_write, cpu_intrpt_per_second, net_txcmptot, nfs_2sd_rename, nfs_4cd_lookup, nfs_2sd_null, nfs_2sd_fsstat, nfs_4sd_rename, nfs_3sd_commit, net_rxcmptot, net_rxmlttot, cpu_l_avg5, nfs_4cd_link, net_txerrstot, cpu_l_avg1, net_txkbtot, date, nfs_3sd_read, nfs_3cd_getattr, nfs_3cd_remove, nfs_3cd_symlink, cpu_soft_percent, nfs_2cd_remove, time, nfs_2cd_fsstat, nfs_3cd_setattr, nfs_4cd_null) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)'] [parameters: ((0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 822.0, '2016-02-26 10:11:24.502000', 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 291.0, 0.0, 0.0, 0.0, u'jterator.init', 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 386.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.04, 0.0, 0.0, 0.0, 0.08, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, u'compute006', 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.05, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 100.0, 0.0, 0.0, 0.0, 0.0, 0.0), (0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 596.0, '2016-02-26 10:11:24.601000', 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 291.0, 0.0, 0.0, 0.0, u'jterator.init', 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 323.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.04, 0.0, 0.0, 0.0, 0.08, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, u'compute006', 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.05, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 100.0, 0.0, 0.0, 0.0, 0.0, 0.0), (0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 406.0, '2016-02-26 10:11:24.702000', 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 291.0, 0.0, 0.0, 0.0, u'jterator.init', 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 2.0, 0.0, 188.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.04, 0.0, 0.0, 0.0, 0.08, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, u'compute006', 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.05, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 100.0, 0.0, 0.0, 0.0, 0.0, 0.0), (0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 525.0, '2016-02-26 10:11:24.801000', 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 0.0, 0.0, 291.0, 0.0, 0.0, 0.0, u'jterator.init', 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 283.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.04, 0.0, 0.0, 0.0, 0.08, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, u'compute006', 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.05, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 99.0, 0.0, 0.0, 0.0, 0.0, 0.0), (0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 700.0, '2016-02-26 10:11:24.901000', 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 291.0, 0.0, 0.0, 0.0, u'jterator.init', 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 300.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.04, 0.0, 0.0, 0.0, 0.08, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, u'compute006', 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.05, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 100.0, 0.0, 0.0, 0.0, 0.0, 0.0), (0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 570.0, '2016-02-26 10:11:25.001000', 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 291.0, 0.0, 0.0, 0.0, u'jterator.init', 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 250.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.04, 0.0, 0.0, 0.0, 0.08, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, u'compute006', 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.05, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 100.0, 0.0, 0.0, 0.0, 0.0, 0.0), (0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 580.0, '2016-02-26 10:11:25.101000', 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 291.0, 0.0, 0.0, 0.0, u'jterator.init', 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 300.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.04, 0.0, 0.0, 0.0, 0.08, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, u'compute006', 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.05, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 100.0, 0.0, 0.0, 0.0, 0.0, 0.0), (0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 720.0, '2016-02-26 10:11:25.201000', 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 291.0, 0.0, 0.0, 0.0, u'jterator.init', 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 330.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.03, 0.0, 0.0, 0.0, 0.08, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, u'compute006', 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.05, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 100.0, 0.0, 0.0, 0.0, 0.0, 0.0)  ... displaying 10 of 51 total bound parameter sets ...  (0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 330.0, '2016-02-26 10:11:29.401000', 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 295.0, 0.0, 0.0, 0.0, u'jterator.init', 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 150.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.03, 0.0, 0.0, 0.0, 0.08, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, u'compute006', 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.05, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 100.0, 0.0, 0.0, 0.0, 0.0, 0.0), (0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 810.0, '2016-02-26 10:11:29.501000', 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 295.0, 0.0, 0.0, 0.0, u'jterator.init', 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 380.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.03, 0.0, 0.0, 0.0, 0.08, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, u'compute006', 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.05, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 100.0, 0.0, 0.0, 0.0, 0.0, 0.0))]

Is there something I'm missing?  Should I call some kind of sync/finish after `insert_many()`?

**Comment by riccardomurri** [comment]
Apparently this is an issue with the default parameters used with SQLite DBs.

I disabled connection pooling with::

       dataset.connect(db_uri, engine_kwargs={'poolclass':sqlalchemy.pool.StaticPool})

and now I no longer get any error.  My guess is that with the default
`NullPool` class, the second `insert_many` gets a new DB connection
while the first insert is still running, and then it fails because the
DB is locked for writing.

**Comment by pudo** [comment]
Hm, that's a weird issue -- I use various chained insert_many calls but have never run into this issue. Would you think that setting the static pool as a default for SQLite databases would be a good strategy?

**Comment by riccardomurri** [comment]
> Would you think that setting the static pool as a default for SQLite databases would be a good strategy? 

It worked for me.  I really cannot say whether this would be a good default in the general case.

**Comment by pudo** [comment]
It's in 0.6.3, lets see is there's any further feedback.

**Comment by riccardomurri** [comment]
Thanks for quick action on this!
​

**Comment by gregorleban** [comment]
I actually had a problem using the sqlalchemy using the StaticPool in combination with an sqlite database. I've changed it now to sqlalchemy.pool.NullPool and currently don't have issues anymore.

To be more elaborate: I'm using a dataset python package and accessing several sqlite tables. The access to these tables can be from multiple threads. Sometimes (not consistently) I was getting the following exceptions:

    (sqlite3.ProgrammingError) SQLite objects created in a thread can only be used in that same thread.The object was created in thread id 140143768454976 and this is thread id 140143651112704

When I changed the pool class from StaticPool to NullPool, the problem went away.
